builtin/providers/openstack/resource_openstack_lb_pool_v1.go

```diff
@@ -0,0 +1,334 @@
+							Optional: true,
+							ForceNew: true,
+						},
+						"address": &schema.Schema{
+							Type:     schema.TypeString,
+							Required: true,
+							ForceNew: true,
+						},
+						"port": &schema.Schema{
+							Type:     schema.TypeInt,
+							Required: true,
+							ForceNew: true,
+						},
+						"admin_state_up": &schema.Schema{
+							Type:     schema.TypeBool,
+							Required: true,
+							ForceNew: false,
+						},
+					},
+				},
+				Set: resourceLBMemberV1Hash,
+			},
+			"monitor_ids": &schema.Schema{
+				Type:     schema.TypeSet,
+				Optional: true,
+				ForceNew: false,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+				Set: func(v interface{}) int {
+					return hashcode.String(v.(string))
+				},
+			},
+		},
+	}
+}
+
+func resourceLBPoolV1Create(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+	networkingClient, err := config.networkingV2Client(d.Get("region").(string))
+	if err != nil {
+		return fmt.Errorf("Error creating OpenStack networking client: %s", err)
+	}
+
+	createOpts := pools.CreateOpts{
+		Name:     d.Get("name").(string),
+		Protocol: d.Get("protocol").(string),
+		SubnetID: d.Get("subnet_id").(string),
+		LBMethod: d.Get("lb_method").(string),
+		TenantID: d.Get("tenant_id").(string),
+	}
+
+	log.Printf("[INFO] Requesting lb pool creation")
+	p, err := pools.Create(networkingClient, createOpts).Extract()
+	if err != nil {
+		return fmt.Errorf("Error creating OpenStack LB pool: %s", err)
+	}
+	log.Printf("[INFO] LB Pool ID: %s", p.ID)
+
+	d.SetId(p.ID)
+
+	if mIDs := resourcePoolMonitorIDsV1(d); mIDs != nil {
+		for _, mID := range mIDs {
+			_, err := pools.AssociateMonitor(networkingClient, p.ID, mID).Extract()
+			if err != nil {
+				return fmt.Errorf("Error associating monitor (%s) with OpenStack LB pool (%s): %s", mID, p.ID, err)
+			}
+		}
+	}
+
+	if memberOpts := resourcePoolMembersV1(d); memberOpts != nil {
+		for _, memberOpt := range memberOpts {
+			_, err := members.Create(networkingClient, memberOpt).Extract()
+			if err != nil {
+				return fmt.Errorf("Error creating OpenStack LB member: %s", err)
+			}
+		}
+	}
+
+	return resourceLBPoolV1Read(d, meta)
+}
+
+func resourceLBPoolV1Read(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+	networkingClient, err := config.networkingV2Client(d.Get("region").(string))
+	if err != nil {
+		return fmt.Errorf("Error creating OpenStack networking client: %s", err)
+	}
+
+	p, err := pools.Get(networkingClient, d.Id()).Extract()
+	if err != nil {
+		return CheckDeleted(d, err, "LB pool")
+	}
+
+	log.Printf("[DEBUG] Retreived OpenStack LB Pool %s: %+v", d.Id(), p)
+
+	d.Set("region", d.Get("region").(string))
+	d.Set("name", p.Name)
+	d.Set("protocol", p.Protocol)
+	d.Set("subnet_id", p.SubnetID)
+	d.Set("lb_method", p.LBMethod)
+
+	if t, exists := d.GetOk("tenant_id"); exists && t != "" {
+		d.Set("tenant_id", p.TenantID)
+	} else {
+		d.Set("tenant_id", "")
+	}
```

Comment:
Here's a combo of both extra-guard and no-guard styles. What's the difference in behavior you expect here?